zebra-state/src/service/chain_tip.rs

```diff
@@ -310,7 +311,7 @@ impl ChainTipChange {
     ///
     /// If a lot of blocks are committed at the same time,
     /// the change will skip some blocks, and return a [`Reset`].
-    pub async fn tip_change(&mut self) -> Result<TipAction, watch::error::RecvError> {
+    async fn tip_change(&mut self) -> Result<TipAction, watch::error::RecvError> {
```

Comment:
### Other PRs

I'll need this method in #2729, because block gossip is an async task that waits for state changes.

But we should definitely rename it to avoid confusion.

```suggestion
    pub async fn wait_for_tip_change(&mut self) -> Result<TipAction, watch::error::RecvError> {
```